### PR TITLE
WIP: Marketplace approval integration and general fixes

### PR DESCRIPTION
**Pull request by mservidio**
- Fixed: Error message for performMetadataUpdate not written out to log
- Fixed: Error message for save account and save policy aren't displayed to user
- Fixed: Error message for admin operations aren't displayed to user 
- Fixed: View sync issue where expiry isn't defined
- Create Procurement Approvals UI
- Add marketplaceSolutionId and marketplacePlanId to policy objects
- Add UI validation for solutionId and planId in policy form
- Added validation to ensure marketplaceSolutionId and marketplacePlanId are unique to a policy

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

**Comment by mservidio** [comment]
@phriscage Will we be able to remove the library from this? I think based on the size of this PR, let's review and do what we need to merge this into master and then work off of smaller PRs going forward. Thoughts?

**Comment by phriscage** [comment]
> @phriscage Will we be able to remove the library from this? I think based on the size of this PR, let's review and do what we need to merge this into master and then work off of smaller PRs going forward. Thoughts?

Yes. That works. The old library was removed in the last [commit](https://github.com/GoogleCloudPlatform/datashare-toolkit/pull/136/commits/749a36566d1149e1c1490aa568c042a66cc193d2)

File: api/v1alpha/src/accounts/dataManager.js [hunk 1]
```diff
@@ -313,9 +346,170 @@ async function deleteAccount(projectId, accountId, data) {
...
+    // https://cloud.google.com/marketplace/docs/partners/integrated-saas/frontend-integration#verify-jwt
+    const jwt = require('jsonwebtoken');
+
+    /*
+        1. Verify that the JWT signature is using the public key from Google.
+        2. Verify that the JWT has not expired, by checking the exp claim.
+        3. Verify that aud claim is the correct domain for your solution.
+        4. Verify that the iss claim is https://www.googleapis.com/robot/v1/metadata/x509/cloud-commerce-partner@system.gserviceaccount.com
+        5. Verify that sub is not empty.
+    */
```
Comment (phriscage) [inline api/v1alpha/src/accounts/dataManager.js:364]: Can we assume that the JWT signature has been validated from the Istio ingress gateway now and just decode the JWT for the payload?

File: api/v1alpha/src/index.js [hunk 2]
```diff
@@ -99,6 +101,8 @@ const policies = require('./policies/index');
 const accounts = require('./accounts/index');
 // Import the CDS API Admin service router
 const admin = require('./admin/index');
+// Import the CDS API Procurement service router
+const procurements = require('./procurements/index');
 
 /************************************************************
   API Endpoints
```
Comment (phriscage) [inline api/v1alpha/src/index.js:108]: I think we can remove the `router.all('*', checkIfAuthenticated);` [line](https://github.com/GoogleCloudPlatform/datashare-toolkit/pull/136/files#diff-bb5d23004b97fd9d4fb239ca7976122fL118) now that Istio provides the AuthN & AuthZ

Comment (mservidio) [reply api/v1alpha/src/index.js:108]: Will it be able to handle the regex's?

`if (req.path.startsWith('/docs/')) {
        return next();
    } else if (req.path.match(/\/projects\/.+\/accounts:register/g)) {
        // Allow unauthenticated calls to the accounts:register endpoint
        console.log(accounts:register called with headers: ${JSON.stringify(req.headers)} and body: ${JSON.stringify(req.body)});
        return next();
    } else if (req.path.match(/\/projects\/.+\/products/g)) {
        // Allow unauthenticated calls to the accounts:register endpoint
        console.log(products called with headers: ${JSON.stringify(req.headers)} and body: ${JSON.stringify(req.body)});
        return next();
    }`

Comment (phriscage) [reply api/v1alpha/src/index.js:106]: Yes. Updated accordingly:
api/v1alpha/istio-manifests/1.4/authz/allow-data-consumers-policy.tmpl.yaml
api/v1alpha/istio-manifests/1.4/authz/allow-marketplace-service-account-policy.tmpl.yaml

**Pull request merged**